protocol/cache.go

```diff
@@ -0,0 +1,42 @@
+package protocol
+
+import (
+	"sync"
+
+	"github.com/golang/groupcache/lru"
+
+	"chain/protocol/bc"
+)
+
+// maxCachedValidatedTxs is the max number of validated txs to cache.
+const maxCachedValidatedTxs = 1000
+
+type prevalidatedTxsCache struct {
+	mu  sync.Mutex
```

Comment:
D'oh! My bad. It is a little unfortunate that this code shows up as changed in the diff.